tests/app_tests/endpoint_tests/posts_test.go

```diff
@@ -64,11 +64,11 @@ func TestPostFailureStringKey(t *testing.T) {
 
 	router.ServeHTTP(responseRecorder, request)
 
-	assert.Equal(t, http.StatusInternalServerError, responseRecorder.Code)
+	assert.Equal(t, http.StatusBadRequest, responseRecorder.Code)
```

Comment:
What I meant was to leave these tests as `assert.Equal(t, http.StatusInternalServerError, responseRecorder.Code)`, but then we create a different PR to change the return error here!

Also, I will suggest changing these lines to `require.Equal(t, http.StatusInternalServerError, responseRecorder.Code)` - The require module will make the test fail straightaway, while `assert` will continue running the test even after the expected value was wrong.